src/main/java/killjoy/processing/Parser.java

```diff
@@ -0,0 +1,139 @@
+package killjoy.processing;
+
+import killjoy.task.Task;
+import killjoy.main.UserInterface;
+
+import java.time.LocalDateTime;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the Parser class of the KillJoy application.
+ * Contains methods to parse user input.
+ */
+public class Parser {
+    private static UserInterface ui;
+
+    public Parser(UserInterface ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user input to determine the type of task.
+     * @param input The user input.
+     * @return The type of task.
+     */
+    public static Task.TaskType parseUserInput(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        String[] inputSplitBySpace = inputSplitBySlash[0].split(" ");
+        String typeOfTask = inputSplitBySpace[0];
+
+        try {
+            return Task.TaskType.valueOf(typeOfTask.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Parses the user input to determine the description of the task.
+     * @param input The user input.
+     * @return The description of the task.
+     */
+    public static String getDescription(String input) {
+        String[] inputSplitBySlash = input.split("/");
+        return inputSplitBySlash[0].replaceFirst("todo ", "")
+                .replaceFirst("deadline ", "").replaceFirst("event ", "");
+    }
+
+    /**
+     * Parses the user input to determine the by date of the task.
+     * @param input The user input.
+     * @return The by date of the task.
+     */
+    public static String getBy(String input) {
```

Comment:
i think the naming of the method could be something clearer to prevent ambiguity!